### PR TITLE
ENH add migration for gcc 9.3.0

### PR DESCRIPTION
**Pull request by beckermr**
Checklist
* [ ] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

File: recipe/migrations/gcc930.yaml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    version
+  migration_number: 1
+  build_number: 1
+  override_cbc_keys:
+    - fortran_compiler_stub
+
+c_compiler_version:            # [unix]
+  - 11                         # [osx and arm64]
+  - 10                         # [osx and x86_64]
+  - 9.3.0                      # [linux64]
```
Comment (isuruf) [inline recipe/migrations/gcc930.yaml:14]:
```suggestion
  - 9                      # [linux64]
```

File: recipe/migrations/gcc930.yaml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  - 9.3.0                      # [aarch64]
+  - 9.3.0                      # [ppc64le]
+  - 8.3                        # [armv7l]
```
Comment (isuruf) [inline recipe/migrations/gcc930.yaml:17]:
```suggestion
```

File: recipe/migrations/gcc930.yaml (hunk 1 above)
Comment (beckermr) [reply recipe/migrations/gcc930.yaml:14]: Should they all be `9`?

Comment (isuruf) [reply recipe/migrations/gcc930.yaml:14]: yes

**Comment by beckermr** [comment]
This is waiting on the rebuild of gfortran on osx so that the hashes in the compilers are unique.

**Comment by isuruf** [comment]
Aren't we going to build both?

**Comment by isuruf** [comment]
Can you also clean up the versions in `conda_build_config.yaml`?

**Comment by beckermr** [comment]
Build both versions of gfortran?

**Comment by beckermr** [comment]
> Aren't we going to build both?

I still don't follow this question @isuruf. 

I think I got the rest of the changes.

**Comment by isuruf** [comment]
I was under the impression that this will take a long time as well and that we should build both versions (7 and 9) to not interfere other migrations.

**Comment by beckermr** [comment]
Ahhhh. I can put back the version 7 on osx and linux. We don't need it on aarch or ppc since the SONAME is the same.

**Comment by isuruf** [comment]
aarch64 used 7.*. For ppc64le, we still need to rebuild for the new run_exports to work (unless we do a repodata patch).

**Comment by beckermr** [comment]
@conda-forge/core as currently set up, this migration will remove any compiler pins related to armv7l builds. Do we need those platforms at all?

**Comment by beckermr** [comment]
> aarch64 used 7.*. For ppc64le, we still need to rebuild for the new run_exports to work (unless we do a repodata patch).

Right. I think I got the old 7 platforms correct. I can add 8 for popc64le

**Comment by jjhelmus** [comment]
> @conda-forge/core as currently set up, this migration will remove any compiler pins related to armv7l builds. Do we need those platforms at all?

The `armv7l` platform can be removed, it is not being built.  If someone wants to add this platform to conda-forge in the future they can re-add the correct pins.

**Comment by beckermr** [comment]
great @jjhelmus! I think this is ready to merge then!

**Comment by beckermr** [comment]
Any other comments here @conda-forge/core?

**Comment by beckermr** [comment]
announcement here: https://github.com/conda-forge/conda-forge.github.io/pull/1166

**Pull request merged**